From 6411f6b8c8532fc81a655b793f1e54ddaad9c9c4 Mon Sep 17 00:00:00 2001
From: Himanshu <himanshu@cloudera.com>
Date: Fri, 30 Mar 2012 15:58:46 -0600
Subject: [PATCH 013/151] HBASE-5655 Cap space usage of default log4j rolling policy

Reason: Improvement
Author: Himanshu Vashishtha
Ref: CDH-3602 (sub task: CDH-5069)

	modified:   bin/hbase-daemon.sh
	modified:   conf/hbase-env.sh
	modified:   conf/log4j.properties
---
 bin/hbase-daemon.sh   |    2 +-
 conf/hbase-env.sh     |    8 ++++++++
 conf/log4j.properties |   13 +++++++++++++
 3 files changed, 22 insertions(+), 1 deletions(-)

diff --git a/bin/hbase-daemon.sh b/bin/hbase-daemon.sh
index 22c2284..c22e2f5 100755
--- a/bin/hbase-daemon.sh
+++ b/bin/hbase-daemon.sh
@@ -115,7 +115,7 @@ if [ "$JAVA_HOME" = "" ]; then
 fi
 JAVA=$JAVA_HOME/bin/java
 export HBASE_LOGFILE=hbase-$HBASE_IDENT_STRING-$command-$HOSTNAME.log
-export HBASE_ROOT_LOGGER="INFO,DRFA"
+export HBASE_ROOT_LOGGER=${HBASE_ROOT_LOGGER:-"INFO,RFA"}
 logout=$HBASE_LOG_DIR/hbase-$HBASE_IDENT_STRING-$command-$HOSTNAME.out  
 loglog="${HBASE_LOG_DIR}/${HBASE_LOGFILE}"
 pid=$HBASE_PID_DIR/hbase-$HBASE_IDENT_STRING-$command.pid
diff --git a/conf/hbase-env.sh b/conf/hbase-env.sh
index f7856b1..30139c0 100644
--- a/conf/hbase-env.sh
+++ b/conf/hbase-env.sh
@@ -79,3 +79,11 @@ export HBASE_OPTS="-XX:+UseConcMarkSweepGC"
 
 # Tell HBase whether it should manage it's own instance of Zookeeper or not.
 # export HBASE_MANAGES_ZK=true
+# The default log rolling policy is RFA, where the log file is rolled as per the size defined for the 
+# RFA appender. Please refer to the log4j.properties file to see more details on this appender.
+# In case one needs to do log rolling on a date change, one should set the environment property
+# HBASE_ROOT_LOGGER to "<DESIRED_LOG LEVEL>,DRFA".
+# For example:
+# HBASE_ROOT_LOGGER=INFO,DRFA
+# The reason for changing default to RFA is to avoid the boundary case of filling out disk space as 
+# DRFA doesn't put any cap on the log size. Please refer to HBase-5655 for more context.
diff --git a/conf/log4j.properties b/conf/log4j.properties
index 0010d6a..9ff1c04 100644
--- a/conf/log4j.properties
+++ b/conf/log4j.properties
@@ -28,6 +28,19 @@ log4j.appender.DRFA.layout.ConversionPattern=%d{ISO8601} %p %c: %m%n
 # Debugging Pattern format
 #log4j.appender.DRFA.layout.ConversionPattern=%d{ISO8601} %-5p %c{2} (%F:%M(%L)) - %m%n
 
+# Rolling File Appender properties
+hbase.log.maxfilesize=256MB
+hbase.log.maxbackupindex=20
+
+# Rolling File Appender
+log4j.appender.RFA=org.apache.log4j.RollingFileAppender
+log4j.appender.RFA.File=${hbase.log.dir}/${hbase.log.file}
+
+log4j.appender.RFA.MaxFileSize=${hbase.log.maxfilesize}
+log4j.appender.RFA.MaxBackupIndex=${hbase.log.maxbackupindex}
+
+log4j.appender.RFA.layout=org.apache.log4j.PatternLayout
+log4j.appender.RFA.layout.ConversionPattern=%d{ISO8601} %p %c: %m%n
 
 #
 # console
-- 
1.7.0.4

